components/server/src/user/user-service.ts

```diff
@@ -108,13 +108,18 @@ export class UserService {
         }
     }
 
-    private cachedIsFirstUser: true | undefined = undefined;
+    private cachedIsFirstUser: boolean | undefined = undefined;
     public async createUser({identity, token, userUpdate }: CreateUserParams): Promise<User> {
         log.debug('Creating new user.', { identity, 'login-flow': true });
 
         const prevIsFirstUser = this.cachedIsFirstUser;
-        this.cachedIsFirstUser = true; // avoid races
-        const isFirstUser = prevIsFirstUser || (await this.userDb.getUserCount() === 0);
+        this.cachedIsFirstUser = false; // as we're going to create one
```

Comment:
I understand the code, but not the comment. How are we not the first user if there isn't one yet?

Reply:
it's the value for `cachedIsFirstUser` which is read on next call of the async `createUser`.

the comment "we are about to create one" isn't great, though. just because this is async, setting the value immediately should be right. checking with previous value, or awaiting user count for comparison can follow then.

Reply:
comment is updated.